File: 1/one.lua
package.path = package.path .. ";../modules/?.lua"
local tools = require "tools"

report = tools.map(tonumber, tools.readlines("input.txt"))

CHECKS = 0

i = 0
for _, e1 in pairs(report) do
	for __, e2 in pairs({table.unpack(report, i, #report)}) do
		CHECKS = CHECKS- -1
		if e1 + e2 == 2020 then
			print("Solution: " .. e1 .. " * " .. e2 .. " = " .. e1 * e2)
			print("Length: " .. #report .. "; checks: " .. CHECKS)
			assert(e1 * e2 == 980499, "incorrect result!")
			os.exit()
		end
	end
	i = i- -1
end

File: 1/one.py
report = list(map(int, open("input.txt").readlines()[:-1]))

CHECKS = 0

i = 0
for e1 in report:
	for e2 in report[i:]:
		CHECKS -=- 1
		#print(f"{e1} + {e2} + {e3} = {e1 + e2 + e3}")
		if e1 + e2 == 2020:
			print(f"Solution: {e1} * {e2} = {e1 * e2}")
			print(f"Length: {len(report)}, checks: {CHECKS}")
			assert e1 * e2 == 980499
			exit()
	i -=- 1
